.github/workflows/CI.yml

```diff
@@ -109,28 +109,42 @@ jobs:
       - name: Run build
         run: |
           chown -R 1000:1000 `pwd`
-          su `id -un 1000` -c "./gradlew precommit --parallel && ./gradlew jacocoTestReport && cp -v build/reports/jacoco/test/jacocoTestReport.xml ./jacocoTestReport.xml"
+          su `id -un 1000` -c "./gradlew precommit -x integTest --parallel"
 
-      - uses: actions/upload-artifact@v4
-        if: ${{ matrix.os }} == "ubuntu-latest"
-        with:
-          name: coverage-report-${{ matrix.os }}-${{ matrix.java }}
-          path: ./jacocoTestReport.xml
-
-  Precommit-codecov:
+  integTest:
     needs: Precommit-neural-search-linux
     strategy:
       matrix:
-        java: [21, 24]
-        os: [ubuntu-latest]
+        java: [ 21, 23 ]
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
```

Comment:
do we need all 3 platforms here? I think only linux is sufficient, others might not provide significant additional coverage

Reply:
Kept all 3 to understand if integ tests passes on all 3 env. For multi node changed it to only linux

Reply:
Updated to just linux